cloud: add changefeed docs for TiDB Cloud Essential (#22244)
pingcap/docs: tidb-cloud/essential-changefeed-overview.md

---
title: Changefeed (Beta)
summary: TiDB Cloud changefeed helps you stream data from TiDB Cloud to other data services.
---

# Changefeed (Beta)

TiDB Cloud changefeed helps you stream data from TiDB Cloud to other data services. Currently, TiDB Cloud supports streaming data to Apache Kafka and MySQL.

> **Note:**
>
> - Currently, TiDB Cloud only allows up to 10 changefeeds per {{{ .essential }}} cluster.
> - For [{{{ .starter }}}](/tidb-cloud/select-cluster-tier.md#starter) clusters, the changefeed feature is unavailable.

## Supported regions

> The changefeed feature is currently supported in the following regions:

Reviewer: ```suggestion
The changefeed feature is available in the following regions:
```

| Cloud Provider | Supported Regions |
| --- | --- |
| Alibaba Cloud | ap-southeast-1<br>cn-hongkong<br>ap-southeast-5 |

> | AWS | us-east-1 |

Reviewer 2: ```suggestion
| Alibaba Cloud | `ap-southeast-1`<br>`cn-hongkong`<br>`ap-southeast-5` |
| AWS          | `us-east-1` |
```

Additional regions will be supported in the future. For immediate support in a specific region, contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md).

## View the Changefeed page

To access the changefeed feature, take the following steps:

1. Log in to the [TiDB Cloud console](https://tidbcloud.com/) and navigate to the [**Clusters**](https://tidbcloud.com/project/clusters) page of your project.

    > **Tip:**
    >
    > You can use the combo box in the upper-left corner to switch between organizations, projects, and clusters.

2. Click the name of your target cluster to go to its overview page, and then click **Data** > **Changefeed** in the left navigation pane. The changefeed page is displayed.

> On the **Changefeed** page, you can create a changefeed, view a list of existing changefeeds, and operate the existing changefeeds (such as scaling, pausing, resuming, editing, and deleting a changefeed).

Author: ```suggestion
On the **Changefeed** page, you can create a changefeed, view a list of existing changefeeds, and operate the existing changefeeds (such as pausing, resuming, editing, and deleting a changefeed).
```

## Create a changefeed

To create a changefeed, refer to the tutorials:

- [Sink to Apache Kafka](/tidb-cloud/essential-changefeed-sink-to-kafka.md)
- [Sink to MySQL](/tidb-cloud/essential-changefeed-sink-to-mysql.md)

## View a changefeed

You can view a changefeed using the TiDB Cloud console or the TiDB Cloud CLI.

<SimpleTab>
<div label="Console">

1. Navigate to the [**Changefeed**](#view-the-changefeed-page) page of your target TiDB cluster.
2. Locate the corresponding changefeed you want to view, and click **...** > **View** in the **Action** column.
3. You can see the details of a changefeed, including its configuration, status, and metrics.

</div>

<div label="CLI">

```
ticloud serverless changefeed get -c <cluster-id> --changefeed-id <changefeed-id>
```

</div>
</SimpleTab>

## Pause or resume a changefeed

You can pause or resume a changefeed using the TiDB Cloud console or the TiDB Cloud CLI.

<SimpleTab>
<div label="Console">

1. Navigate to the [**Changefeed**](#view-the-changefeed-page) page of your target TiDB cluster.

> 2. Locate the corresponding changefeed you want to pause or resume. In the **Action** column, click **...** > **Pause/Resume**.

Reviewer: ```suggestion
2. Locate the corresponding changefeed you want to pause or resume, and click **...** > **Pause/Resume** in the **Action** column.
```

</div>

<div label="CLI">

> To pause a changefeed:

Reviewer: ```suggestion
To pause a changefeed, run the following command:
```

```
ticloud serverless changefeed pause -c <cluster-id> --changefeed-id <changefeed-id>

> ```

Reviewer: ````suggestion
```bash
ticloud serverless changefeed pause --cluster-id <cluster-id> --changefeed-id <changefeed-id>
```
````

To resume a changefeed:

```
ticloud serverless changefeed resume -c <cluster-id> --changefeed-id <changefeed-id>
```

</div>
</SimpleTab>

## Edit a changefeed

> **Note:**
>
> TiDB Cloud currently only allows editing changefeeds that are in the `Paused` state.

You can edit a changefeed using the TiDB Cloud console or the TiDB Cloud CLI.

<SimpleTab>
<div label="Console">

1. Navigate to the [**Changefeed**](#view-the-changefeed-page) page of your target TiDB cluster.
2. Locate the changefeed you want to pause. In the **Action** column, click **...** > **Pause**.
3. When the changefeed status changes to `Paused`, click **...** > **Edit** to edit the corresponding changefeed.

    TiDB Cloud populates the changefeed configuration by default. You can modify the following configurations:

>     - Apache Kafka sink: all configurations except **Destination**, **Connection** and **Start Position**

Reviewer: ```suggestion
    - Apache Kafka sink: all configurations except **Destination**, **Connection**, and **Start Position**
```

    - MySQL sink: all configurations except **Destination**, **Connection** and **Start Position**

4. After editing the configuration, click **...** > **Resume** to resume the corresponding changefeed.

</div>

<div label="CLI">

Edit a changefeed with Apache Kafka sink:

```
ticloud serverless changefeed edit -c <cluster-id> --changefeed-id <changefeed-id> --name <new-displayName> --kafka <full-specified-kafka> --filter <full-specified-filter>

> ```

Reviewer: ````suggestion
```bash
ticloud serverless changefeed edit --cluster-id <cluster-id> --changefeed-id <changefeed-id> --name <new-displayName> --kafka <full-specified-kafka> --filter <full-specified-filter>
```
````

Edit a changefeed with MySQL sink:

```
ticloud serverless changefeed edit -c <cluster-id> --changefeed-id <changefeed-id> --name <new-displayName> --mysql <full-specified-mysql> --filter <full-specified-filter>

> ```

Reviewer: ````suggestion
```bash
ticloud serverless changefeed edit --cluster-id <cluster-id> --changefeed-id <changefeed-id> --name <new-displayName> --mysql <full-specified-mysql> --filter <full-specified-filter>
```
````
